workitem/link/type_repository.go

```diff
@@ -168,48 +169,72 @@ func (r *GormWorkItemLinkTypeRepository) Save(ctx context.Context, modelToSave W
 	return &modelToSave, nil
 }
 
-func (r *GormWorkItemLinkTypeRepository) ListSourceLinkTypes(ctx context.Context, witID uuid.UUID) ([]WorkItemLinkType, error) {
+// ListSourceLinkTypes returns all the link types where the given work item type
+// can be used in the source of the link type. It does so by analysing the type
+// combinations that are associated with the given link type.
+func (r *GormWorkItemLinkTypeRepository) ListSourceLinkTypes(ctx context.Context, spaceID, witID uuid.UUID) ([]WorkItemLinkType, error) {
 	db := r.db.Model(WorkItemLinkType{})
-	query := fmt.Sprintf(`
+	where := fmt.Sprintf(`
 			-- Get link types we can use with a specific WIT if the WIT is at the
 			-- source of the link.
-			(SELECT path FROM %[2]s WHERE id = %[1]s.source_type_id LIMIT 1)
-			@>
-			(SELECT path FROM %[2]s WHERE id = ? LIMIT 1)`,
-		WorkItemLinkType{}.TableName(),
+			space_id IN (? /* The current space */, ? /* The system space (NOTE: Keep this even with space templates!!!) */)
```

Comment:
Let's separate the comments from the query itself. Otherwise it is too difficult to understand what this query does. So either put the comments above or as golang comments before the query.